app/scripts/lib/modal_manager.js

```diff
@@ -0,0 +1,127 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'underscore',
+  'jquery',
+  'velocity',
+  'velocityui'
+], function (_, $) {
+  'use strict';
+
+  function ModalManager () {
+    this._viewStates = [];
+    this.$modal = $('#modal');
+
+    // Listen for all clicks
+    $(document).on('click', _.bind(this._closeOnClick, this));
+
+    // Close modal on resize to prevent weirdness between desktop and mobile
+    $(document).on('resize', _.debounce(_.bind(this.close, this), 100));
+
+    // Handle default button actions
+    this.$modal.on('click', 'a.close', _.bind(this.close, this));
+    this.$modal.on('click', 'a.back', _.bind(this.pop, this));
+  }
+
+  _.extend(ModalManager.prototype, {
+    open: function (view, options) {
+      this._destroyViews();
+
+      this.push(view, options);
+    },
+
+    push: function (view, options) {
+      this._viewStates.push({ view: view, options: (options || {}) });
+
+      this._show();
+    },
+
+    pop: function () {
+      var viewState = this._viewStates.pop();
+
+      if (viewState) {
+        viewState.view.destroy();
+      }
+
+      if (this._viewStates.length > 0) {
+        this._show();
+      } else {
+        this._hide();
+      }
+    },
+
+    close: function (options) {
+      if (!options) {
+        options = {};
+      }
```

Comment:
FWIW: We do `if (!options) {...}` here and the shortcut `options: (options || {})` on line 36. Not sure if we should standardize, or whatever. I just calls em like i sees em.